### PR TITLE
8345127: Add --sun-misc-unsafe-memory-access=allow when running tests until JDK-8334137 is fixed

### PR DESCRIPTION
**Pull request by kevinrushforth**
Add the `--sun-misc-unsafe-memory-access=allow` flag until [JDK-8334137](https://bugs.openjdk.org/browse/JDK-8334137) is fixed to avoid warnings when running tests (automated and manual) with JDK 24.

I've tested this using a boot JDK of JDK 22.0.2, JDK 23 (the boot JDK we use in production), and in our CI with the latest promoted build of JDK 24, which now warns by default when a deprecated method in `sun.misc.Unsafe` is called.

Without this fix, we see a warning when running our automated or manual tests with JDK 24. With this fix, we see no warnings.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8345127](https://bugs.openjdk.org/browse/JDK-8345127): Add --sun-misc-unsafe-memory-access=allow when running tests until JDK-8334137 is fixed (**Bug** - P3)

### Reviewers
 * [Ambarish Rapte](https://openjdk.org/census#arapte) (@arapte - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1650/head:pull/1650` \
`$ git checkout pull/1650`

Update a local copy of the PR: \
`$ git checkout pull/1650` \
`$ git pull https://git.openjdk.org/jfx.git pull/1650/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1650`

View PR using the GUI difftool: \
`$ git pr show -t 1650`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1650.diff">https://git.openjdk.org/jfx/pull/1650.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1650#issuecomment-2541800632)
</details>

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -764,6 +767,7 @@ try (BufferedReader inStream = new java.io.BufferedReader(new java.io.InputStrea
             // should the need arise.
 //            ext.jdk18OrLater = jdkVersionInfo.feature() >= 18
 
+            ext.jdk23OrLater = jdkVersionInfo.feature() >= 23
             ext.jdk24OrLater = jdkVersionInfo.feature() >= 24
```
Comment (kevinrushforth) [inline build.gradle:771]: We really only need to add `--sun-misc-unsafe-memory-access=allow` when running with JDK 24 or later ("allow" is the default in JDK 23). I want to do initial testing on JDK 23 with this flag, but will likely change all of the checks to `jdk24OrLater` for the final version, in which case we won't need the `jdk23OrLater` flag.

Comment (kevinrushforth) [reply build.gradle]: I made the above change.

**Comment by kevinrushforth** [comment]
@arapte can you review this?

**Review by arapte — APPROVED** [review]
LGTM, verified with 24 ea build28

**Pull request closed without merging**